src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -33,18 +33,18 @@ class thread_state {
     long task_id = -1;
 };
 
-template<typename T>
 class rollback {
+    using T = const std::function<void()>;
```

Comment:
sorry could this be not T also? e.g. `callback_type` or just use `std::function` directly (since T is normally reserved for a template argument, and this isn't a template argument now.